src/components/kalman_component/kalman.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Functions available in this file include:
+#   - setup_kf
+#   - run_kf
+
+# Description: Setup Kalman filter prereqiuisites
+# Usage:
+#   setup_kf
+setup_kf() {
+    StateVectorFile="${RunDirs}/StateVector.nc"
+
+    # Create a parent directory for the Kalman filter inversions
+    # Include a link to the state vector file for use with run_inversion.sh
+    mkdir -p ${RunDirs}/kf_inversions
+    ln -sf $StateVectorFile ${RunDirs}/kf_inversions/StateVector.nc
+
+    # copy kf notebook to kf_inversions directory
+    cp ${InversionPath}/src/notebooks/kf_notebook.ipynb ${RunDirs}/kf_inversions/
+
+    # Define Kalman filter update periods
+    python ${InversionPath}/src/components/kalman_component/make_periods_csv.py $StartDate $EndDate $UpdateFreqDays $RunDirs; wait
+
+    # Create unit scale factor file
+    python ${InversionPath}/src/components/kalman_component/make_unit_sf.py $StateVectorFile $RunDirs; wait
+
+    # Create directory to archive prior scale factors for each inversion period
+    mkdir -p ${RunDirs}/archive_sf
+
+    # Number of state vector elements
+    nElements=$(ncmax StateVector ${StateVectorFile})
```

Comment:
You wrote a script to replace `ncmax`, right?

Reply:
Yup, but my replacement shares the same name and formatting as the old one -- so this is correct!